### DISCUSSION TITLE
Question: Is it possible to generate an "Unreleased" section in the changelog?

### DISCUSSION THREAD

**Discussion by arkadioz** [semantic-release/semantic-release#3133] [Release workflow] [answered]
Hi, in my case it would be a good feature to somehow have only the changelog.md updated when executing npx semantic-release from for example feature branches or enhancements, that would only generate the new changelog and update its "Unreleased" section, similar to what this standard shows: https://keepachangelog.com/en/1.1.0/

What I imagine is that when npx semantic-release executes from master/main branch, then it can leave the Unreleased section blank, and fill the new tag section with everything that was on Unreleased, maybe with some kind of parametrization

if it is possible I appreciate any example or documentation to achieve it, thanks

**Comment by travi** [comment, marked as answer]
semantic-release is intended to encourage a continuous deployment style workflow rather than encouraging releasing from feature branches. therefore, it is optimize communication of released features, not unreleased features. there is no current support for highlighting unreleased features and it is unlikely that we would pursue that

**Reply by arkadioz** [reply]
I was discussing this with my team, I believe the unreleased section should not be necessary, we had that section because today the process is manual and it saves time when making a release just replacing that "Unreleased" section with the new version, a valid argument in favor of keeping that section even when the process is upgraded to automatic with the semantic-release tool, maybe would be to say "then how we can keep track of what is pending to go into release, because it takes time before we finally can merge to master or main branches and how do we know all that is pending to be released there?" And I was reading more and found this https://github.com/semantic-release/semantic-release/blob/v23.0.0-beta.1/docs/recipes/release-workflow/pre-releases.md#publishing-pre-releases so my answer/idea would be that, it would be easy to keep track because we would have some branches that would generate pre-release versions like in that example, and if you want you can check the changelog that those pre-release generate and there you could see what is incoming or pending to be released right? So I believe that this Unreleased section should not be necessary.
